app/models/user.rb

```diff
@@ -300,6 +301,16 @@ def update_profile_completed!
     update_profile_completed.save!
   end
 
+  def setup_notification_setting_states!
```

Comment:
I think it would make more sense to have the logic primarily in the `NotificationSetting` model